docs/proposals/27-falco-extension.md

```diff
@@ -0,0 +1,216 @@
+
+## Table of Contents
+
+- [Summary](#summary)
+- [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+- [Alternatives](#alternatives)
+- [References](#references)
+
+## Summary
+
+Falco is a cloud native runtime security tool for Linux operating systems. It
+is designed to detect and alert on abnormal behavior and potential security 
+threats in real-time [1]. We have deployed it in the Gardener landscape and
+can confirm that it is well suited for detecting events that are not ordinary
+runtime events. This includes but is not limited to debugging activities,
+changing host configurations, or installing software in containers. Apart from
+legal debugging activities those types of events most likely indicate malicious
+behavior.
+
+Falco is a powerful runtime threat detection engine but comes with limited
+means to deploy on many Kubernetes clusters or sophisticated event analysis
+and visualization for many clusters. These features are usually provided by
+commercial products with may themselves rely on Falco as the detection
+engine.
+
+This enhancement proposal describes a consistent way for deploying Falco into
+the Gardener infrastructure as well as into shoot clusters operated by Gardener
+users in a "Gardener native" way. This includes managing it, providing Falco
+updates and bringing a central infrastucture for viewing and analyzing events.
+
+## Motivation
+
+While we have invested a lot into Gardener security and carry out regular
+penetration tests we don't have a lot of runtime insights into our Kubernetes
+clusters unless we install 3rd party commercial tools. This includes but is
+not limited to security breaches through misconfigurations or 0day 
+vulnerabilities.
+
+We are planning to close this gap by deploying the Falco runtime security
+tool.
+
+### Goals
+
+- Deploy Falco into the Gardener infrastructure.
+- Allow Gardener users to deploy Falco into their shoot clusters by adding a
+  configuration into their shoot manifests.
+- View and analyze Falco events in the Gardener dashboard.
+- Deploy an optimized ruleset for Gardener managed Kubernetes clusters based 
+  on the Falco rules [5]. Optionally allow deployment of incubating-, and
+  sandbox rules as well as custom rules.
+- Offer multiple Falco versions to users including options for manual- and 
+  automated updates.
+- Allow users to simply turn on Falco in their clusters with a reasonable 
+  default configuration as well as allow power users to fine tune the 
+  configuration (rules, plug-ins)
+
+### Non-Goals
+
+We do not compete with commercial products. While we have plans to develop
+Falco plugins, we will not provide features such as image-, or malware scanning
+that is provided by commercial vendors.
+
+## Proposal
+
+We are planning to provide a Gardener extension that deploys Falco and 
+Falcosidekick into shoot clusters. All events that are generated by Falco are
```

Comment:
maybe a super short description if what the sidekick actually is

, Falco's own event forwarding daemon,